### ISSUE TITLE
Change to rsync options in `Vagrantfile` is ignored until guest reloads 

### ISSUE THREAD

**Issue by stephenreay** [hashicorp/vagrant#11559]
### Vagrant version
2.2.7
### Host operating system
macOS Catalina, 10.15.4
### Guest operating system

N/A

### Vagrantfile

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "koalephant/debian10-amd64"
  config.vm.synced_folder 'foo', '/opt/foo', type: 'rsync', rsync__exclude: ['.environment', '*.options', 'baz']
end
```

### Debug output

https://gist.github.com/stephenreay/c4f18fa1b38bf9927a7ae2f745831e24

### Expected behavior

Current `rsync__exclude` values specified in `Vagrantfile` are passed to `rsync` when running `vagrant rsync`

### Actual behavior

The values `rsync__exclude` values specified in `Vagrantfile` at the time it boots are cached until the next boot of the guest.

### Steps to reproduce

1. Specify some `rsync__exclude` values for a synced folder.
2. Boot the guest
3. Change/Add `rsync__exclude` value(s)
4. Run `vagrant rsync`

I initially encountered this with Parallels, I've since tried VMWare Fusion and Virtualbox, and encountered the same issue.

**Comment by chrisroberts** [comment]
Hi there,

This is the expected behavior of a synced folder. The configuration of a synced folder is defined when bringing a guest up, and modifications to a synced folder are applied by reloading the guest. While rsync is a little different than other synced folder types (like provider native virtual file system mounts or nfs), the behavior is still consistent.

Given that rsync is a bit different than the others, I can see the value in being able to use the current Vagrantfile configuration without requiring a reload of the guest for updates to be applied. So I think the best solution here will be to add a new flag to the rsync command which will allow ignoring the cached folder information within the machine data and read it directly from the Vagrantfile. This will come with some caveats as well though. Adding / removing entries would not be supported, and only a subset of options would be supported for modification.

**Comment by stephenreay** [comment]
I can see why some of it makes sense to apply at up/reload, but yeah, not being able to change the arguments handed to `rsync` when it runs seems a little too restrictive.

Thanks!

**Comment by atwright147** [comment]
Yeah, this is very unintuitive. I have spent the last 3 hours trying to figure this out (luck I found this issue)
